tests/commands/create/test_install_app_support_package.py

```diff
@@ -85,6 +85,17 @@ def test_install_custom_app_support_package_file(create_command, myapp, tmp_path
     assert (support_path / 'internal' / 'file.txt').exists()
 
 
+def test_support_package_url_with_invalid_url(create_command, myapp):
+    "Invalid URL raise MissingNetworkResourceError"
+    # Provide an invalid package URL with unsupported arch
+    url = 'https://briefcase-support.org/python?platform=linux&version=3.6&arch=i686'
+    myapp.support_package = url
+
+    # Confirm that an Exception is Raised
+    with pytest.raises(InvalidSupportPackageURL):
+        create_command.install_app_support_package(myapp)
```

Comment:
From the look of it, this will do a live hit on the support site when the test is run. This is an externality on the test - if, for example, we added i686 support to briefcase-support.org, this test would start failing. The test will also fail if you're not currently connected to a network (e.g., if you're on a plane).

Any externalities should be isolated with mocks so that the test is reliable.

Reply:
Should I provide a dummy url instead, like abc.xyz or should a invalid url structure be also checked?

Reply:
It's not the URL that is the problem - it's the fact that you're calling a network service. Here's the check: turn off wifi on your laptop, and run the test. The test should still pass. 

You need to add a mock for the network access so that *actual* network access isn't required, and we can control the response from the network response to produce a predictable response.

Reply:
Is there some material I can refer learn how to create a mock for the network? Or would I find a similar application in the other tests?

Reply:
We use unittest.Mock as a framework; there's plenty of documentation online for how to use that (including Python's own builtin documentation). The tests for tests/commands/base/test_download_url.py has an example of mocking the requests library.